test/python/utils/mitigation/test_meas.py

```diff
@@ -0,0 +1,707 @@
+                # in list and dict forms using different methods
+                results_dict_1 = meas_filter.apply(results_dict, method="least_squares")
+                results_dict_0 = meas_filter.apply(results_dict, method="pseudo_inverse")
+                results_list_1 = meas_filter.apply(results_list, method="least_squares")
+                results_list_0 = meas_filter.apply(results_list, method="pseudo_inverse")
+
+                # Assert that the results are equally distributed
+                self.assertListEqual(results_list, results_list_0.tolist())
+                self.assertListEqual(results_list, np.round(results_list_1).tolist())
+                self.assertDictEqual(results_dict, results_dict_0)
+                round_results = {}
+                for key, val in results_dict_1.items():
+                    round_results[key] = np.round(val)
+                self.assertDictEqual(results_dict, round_results)
+
+    def test_meas_cal_on_circuit(self):
+        """Test an execution on a circuit."""
+        # Generate the calibration circuits
+        meas_calibs, state_labels, ghz = meas_calib_circ_creation()
+
+        # Run the calibration circuits
+        backend = Aer.get_backend("qasm_simulator")
+        job = qiskit.execute(
+            meas_calibs,
+            backend=backend,
+            shots=self.shots,
+            seed_simulator=SEED,
+            seed_transpiler=SEED,
+        )
+        cal_results = job.result()
+
+        # Make a calibration matrix
+        meas_cal = CompleteMeasFitter(cal_results, state_labels)
+        # Calculate the fidelity
+        fidelity = meas_cal.readout_fidelity()
+
+        job = qiskit.execute(
+            [ghz], backend=backend, shots=self.shots, seed_simulator=SEED, seed_transpiler=SEED
+        )
+        results = job.result()
+
+        # Predicted equally distributed results
+        predicted_results = {"000": 0.5, "111": 0.5}
+
+        meas_filter = meas_cal.filter
+
+        # Calculate the results after mitigation
+        output_results_pseudo_inverse = meas_filter.apply(
+            results, method="pseudo_inverse"
+        ).get_counts(0)
+        output_results_least_square = meas_filter.apply(results, method="least_squares").get_counts(
+            0
+        )
+
+        # Compare with expected fidelity and expected results
+        self.assertAlmostEqual(fidelity, 1.0)
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["000"] / self.shots, predicted_results["000"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_pseudo_inverse["111"] / self.shots, predicted_results["111"], places=1
+        )
+
+        self.assertAlmostEqual(
+            output_results_least_square["111"] / self.shots, predicted_results["111"], places=1
+        )
+
```

Comment:
`test_meas_fitter_with_noise` has gone missing - I assume because we didn't want to migrate the mystery "test_meas_results.json" file, and we're ok because it was more about using an instantiated `CompleteMeasFitter`?